### ISSUE TITLE
Error "Cannot determine type of" for inherited property if child class in separate file

### ISSUE THREAD

**Issue by sergey-dzhus** [python/mypy#20262]
**Bug Report**

In case you have child class, declarated  in separate file, wich has method that uses and then rewrites inherited property, you will get mypy error for this property - Cannot determine type

If child class declarated in same file with base class - no error

**To Reproduce**

1st file code

```python 
class SomeTypeForProperty:
    pass

class A(ABC):
    _a: SomeTypeForProperty | None = None
```

2d file code

```python
class B(A):
    def save(self, a: SomeTypeForProperty) -> bool:
        assert isinstance(self._a, SomeTypeForProperty)

        self._a = a

        return True
```

**Expected Behavior**

Mypy detect type for "_a" from base class

**Actual Behavior**

got error -  Cannot determine type of "_a" 

**Your Environment**

- Mypy version used: 1.18.2
- Python version used: 3.12

**Comment by A5rocks** [comment]
I don't see this behavior:
```sh
a5rocks@whimsy ~/D/mypy> uv run mypy a.py b.py
Success: no issues found in 2 source files
a5rocks@whimsy ~/D/mypy> cat a.py
from abc import ABC

class SomeTypeForProperty:
    pass

class A(ABC):
    _a: SomeTypeForProperty | None = None
a5rocks@whimsy ~/D/mypy> cat b.py
from a import A, SomeTypeForProperty

class B(A):
    def save(self, a: SomeTypeForProperty) -> bool:
        assert isinstance(self._a, SomeTypeForProperty)

        self._a = a

        return True
```

Could you fix the reproducer or tell me what I'm doing wrong?

**Comment by sergey-dzhus** [comment]
I run uv with config file and python files should be inside some dir

uv run mypy --config-file mypy.toml


if files in root dir - no error

#mypy.toml file content
```
[tool.mypy]
python_version = 3.12
files = ["a.py", "b.py"]
cache_dir = "./.cache/mypy"

ignore_missing_imports = true
show_error_codes = true
explicit_package_bases = true
```


if files in some inner dir - get error
#mypy.toml file content
```
[tool.mypy]
python_version = 3.12
files = ["src/a.py", "src/b.py"]
cache_dir = "./.cache/mypy"

ignore_missing_imports = true
show_error_codes = true
explicit_package_bases = true
```

**Comment by A5rocks** [comment]
Can you remove the config options? Specifically these:

```
ignore_missing_imports = true
explicit_package_bases = true
```

Probably one of them is messing things up? (I imagine it's `ignore_missing_imports`...)

**Comment by sergey-dzhus** [comment]
it's explicit_package_bases = true
if I delete it - no error

**Comment by A5rocks** [comment]
The docs for that say:
> This flag tells mypy that top-level packages will be based in either the current directory, or a member of the MYPYPATH environment variable or [mypy_path](https://mypy.readthedocs.io/en/stable/config_file.html#confval-mypy_path) config option. This option is only useful in the absence of `__init__.py`. See [Mapping file paths to modules](https://mypy.readthedocs.io/en/stable/running_mypy.html#mapping-paths-to-modules) for details.

so that makes sense. Try changing `from a import ...` to `from src.a import ...` (I think that will work?). If that works, then that's mypy working as intended. Obviously mypy doesn't know the type of `_a` if it doesn't know what `A` is!

**Comment by sergey-dzhus** [comment]
it works after I added mypy_path = "src"
